todo-app/src/stores/TodoStore.js

```diff
@@ -0,0 +1,108 @@
+import {observable, action, useStrict, runInAction} from 'mobx'
+import Moment from 'moment'
+
+
+useStrict(true)
+
+export default class TodoStore{
+
+    constructor(){
+      this.loadTodos();   // without constructor it won't work, you need to use componentDidMount
+    }
+
+    @observable
+    todos = [];
+
+    @action
+    loadTodos(){
+
+      fetch("https://hyf-react-api.herokuapp.com/todos")
+        .then(response => response.json())
+        .then(response => {
+          // this.setTodos(response)    // if you wanna make the @action tiny [using the function below]
+          console.log("LOAD TODOS : ", response);
+          runInAction(() => {
+            this.todos = response
+          })
+        })
+        .catch(error => {
+          runInAction(() => {
+            console.log(error)
+          })
+        })
+    }
+
+    // @action
+    // setTodos(todos) {
+    //   this.todos = todos
+    // }
+
+    @action  // add a todo item
+    addTodo(text){
+
+      const todo = {
+        "description": text,
+        "deadline": Moment().format('YYYY-MM-DD'),
+        "done": false
+      }
+
+      fetch('https://hyf-react-api.herokuapp.com/todos/create', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(todo), 
+      })
+        .then(response => response.json())
+        .then(response => {
+          console.log("POST RESPONSE   :", response)
+          this.loadTodos();
+        })  
+    }
+
+    @action // change status of a todo
+    onDone(todoID){
+
+      const todo = this.todos.find(todo => todo._id === todoID)
+      if(todo == null) {return}
```

Comment:
Don't write code like this, it's not helpful. First of all, use strict equality checks. Second, if something goes wrong, you should throw an error instead of silently returning the function. The whole point of creating safeguards like this is to create maintainable and debuggable code.